### PR TITLE
Check if user paid for course having free coupon

### LINKED ISSUE #5087: User that's paid for a run but not enrolled end up on order page
This is happening because the logic in `CourseAction` redirects the user to the order summary page if they have a coupon regardless of whether they've already paid for the course or not.

### Steps to Reproduce

- Create a `UserCoupon` for a user
- Attempt to enroll in a course run when edx/devstack is unreachable in order to have enrollment not be created (Order will still be created)
- Attempt to subsequently enroll in the run

### Expected Behavior
- User gets a popup dialog to enroll in the course run

### Actual Behavior
- User ends up on the order summary page

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #5087

#### What's this PR do?
When checking for a free coupon also check if the user has already paid (has fulfilled order) for the course run, and allow them to enroll in the course.

#### How should this be manually tested?
Create a UserCoupon for a user
Attempt to enroll in a course run when edx/devstack is unreachable in order to have enrollment not be created (Order will still be created)
Attempt to subsequently enroll in the run

### CODE REVIEW

**Comment by asadiqbal08** [comment]
@annagav would you like to add a corresponding test use case ?

**Comment by annagav** [comment]
@asadiqbal08 I addressed your comment. Thanks!

File: static/js/components/dashboard/CourseAction_test.js [hunk 1]
```diff
@@ -256,5 +259,29 @@ describe("CourseAction", () => {
       payButton.simulate("click")
       assert.equal(checkoutStub.calledWith(firstRun.course_id), true)
     })
+
+    it("enroll button redirects to order summary if full coupon and order fulfilled", () => {
```
Comment (asadiqbal08) [inline static/js/components/dashboard/CourseAction_test.js:263]: `will redirect to order summary by clicking enroll button if full coupon and order is fulfilled`

**Review by asadiqbal08 — APPROVED** [review]
👍 just a minor nit

**Pull request merged**